### DISCUSSION TITLE
How to configure USB Interface for Arcade buttons

### DISCUSSION THREAD

**Discussion by starburger** [MiczFlor/RPi-Jukebox-RFID#2013] [Q&A] [answered]
Hi everybody
I really need some help with my project, which was going fine until I got to the part where I wanted to configure arcade buttons for controlling my box (I'm building for my 4 year old son).
I bought [this USB arcade interface ](https://www.aliexpress.com/item/1005001700497245.html) and thought I could get it to work by reading through some documentation, but I can't seem to find anything explaing how the USB interface is configured? I've been looking at the gpio_settings.ini file and tried different editing, but can't get anything to work (if it's the right place, how do I know what pins the board/buttons is using?)?

What am I missing? The GPIO Buttons are enabled, when looking at the settings for my Phoniebox.

As you may have guessed, I havn't got a lot of coding/RPi configuration experience to draw from, so any help is highly appriciated.

**Comment by s-martin** [comment, marked as answer]
Hi,

For USB you don’t need to configure GPIO.

please check this documentation https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/develop/components/controls/buttons_usb_encoder/README.md

I’m not sure, if your device will work with this implementation (your link doesn’t work, btw), but you need to try it out.

**Reply by starburger** [reply]
Hi Martin
Thank you for your reply. I clearly missed that part of the documentation.
I tried to follow the directions, but I think some details are missing. I can't find the script in the home directory, neither in the script folder, and, when/if I find it, how exactly do I run it? 

Kind regards, Søren

**Reply by s-martin** [reply]
The readme is a little raw, I improved it a little bit, I hope that helps.

This implementation is fairly generic, so it **should** work with many encoders. Please try it and give feedback, if it works with your encoder, so we could add it to the list of working encoders.

**Reply by starburger** [reply]
It work perfectly, thanks! 

Edited the link in my original post so it works now, but the RPi find the encoder as 'Dragonrise inc. generic usb joystick'. 
If you want to (and can) improve the documentation furter, it would also be helpful to add:

- How to remap, if you make a mistake at the first install. As far as I can tell, it's not posible to just reinstall the script. The input is registeret at second try, but the first configuration is still used afterwards. I have a theory, that I could move the buttons to other unused pins.. but if there aren't any unused, I don't know what to do other than a complete reinstall of phoniebox. But maybe there is an easier fix?
- An explanation of the buttonmap possibilities. I tried to do a translation to some of them. The ones I'm unsure what does, is in **bold**:
'BluetoothToggle': _Turn bluetooth on or off_
'PlayerNext': _Play Next_
'PlayerPause': _Pause_
**'PlayerPauseForce': _Pause even harder?_**
'PlayerPrev': _Previous track_
**'PlayerRandomCard': _Activate (play?) random card?_**
'PlayerRandomFolder': _Play random folder_
'PlayerRandomTrack': _Play random track_
'PlayerSeekBack': _Seek backwards on playing track (**x seconds**)_
'PlayerSeekFarBack': _Seek backwards on playing track (**x seconds**)_
'PlayerSeekFarFwd': _Seek forwards on playing track (**x seconds**)_
'PlayerSeekFwd': _Seek forwards on playing track (**x seconds**)_
'PlayerStop': _Stop playing_
'RecordPlayLatest':  _Play latest recording_
'RecordStart': _Start recording_
'RecordStop': _Stop recording_
'Shutdown': _Shut down the Phoniebox_
'ToggleWifi': _Turn Wifi on or off_
**'Vol0': _Volume mute?_**
'VolD': _Volume down_
'VolU': _Volume up_

**Reply by s-martin** [reply]
Thanks, I added your suggestions to the docs.

**Reply by starburger** [reply]
I'm the one who's thanking :-)

I see you've copy/pasted my translation to the mapping possibilities. I don't know if it would be better to just leave the ones I'm unsure of out completely, instead of my guessing?

Anyway, thanks again for helping.. you've saved me a lot of time!

**Comment by randyesperben** [comment]
Hi!

I'm using future3 and I can't follow this instructions https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/develop/components/controls/buttons_usb_encoder/README.md because I can't find those files.
Any idea?

Many thanks in advance!

**Reply by AlvinSchiller** [reply]
Hi, on future3 these files don't exist, as the new version is a rewrite and works a lot different. So the instructions you mentioned are v2 only.

For future3 this functionality has recently been added to the `future3/develop` branch and will be available in the next release.
See here for instructions:
https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/documentation/builders/event-devices.md

**Reply by randyesperben** [reply]
Thanks @AlvinSchiller ! That makes sense.
I'm using the stable version. How will be the way to update, I will need a fresh install or I can update the new release without do all again?

**Reply by AlvinSchiller** [reply]
A fresh install will be the save call.
For the current state of today a git checkout to switch branches might work as well, but can't guarantee.
